pkg/cli/run.go

```diff
@@ -27,87 +27,185 @@ import (
+	policyType []string
+
+	// IaC type (terraform)
+	iacType string
+
+	// IaC version (for terraform:v12)
+	iacVersion string
+
+	// Path to a single IaC file
+	iacFilePath string
+
+	// Path to a directory containing one or more IaC files
+	iacDirPath string
+
+	// remoteType indicates the type of remote backend. Supported backends are
+	// git s3, gcs, http.
+	remoteType string
+
+	// remoteURL points to the remote Iac repository on git, s3, gcs, http
+	remoteURL string
+
+	// configOnly will output resource config (should only be used for debugging purposes)
+	configOnly bool
+
+	// config file path
+	configFile string
+
+	// the output format for wring the results
+	outputType string
+
+	// UseColors indicates whether to use color output
+	UseColors bool
+	useColors string // used for flag processing
+
+	// Verbose indicates whether to display all fields in default human readlbe output
+	Verbose bool
+}
+
+// Scan executes the terrascan scan command
+func (s *ScanOptions) Scan() error {
+	if err := s.Init(); err != nil {
+		zap.S().Error("scan init failed", zap.Error(err))
+		return err
+	}
+
+	if err := s.Run(); err != nil {
+		zap.S().Error("scan run failed", zap.Error(err))
+		return err
+	}
+	return nil
+}
+
+//Init initalises and validates ScanOptions
+func (s *ScanOptions) Init() error {
+	s.initColor()
+	err := s.validate()
+	if err != nil {
+		zap.S().Error("failed to start scan", zap.Error(err))
+		return err
+	}
+	return nil
+}
+
+// validate config only for human readable output
+// rest command options are validated by the executor
+func (s ScanOptions) validate() error {
+	// human readable output doesn't support --config-only flag
+	// if --config-only flag is set, then exit with an error
+	// asking the user to use yaml or json output format
+	if s.configOnly && strings.EqualFold(s.outputType, humanOutputFormat) {
+		return errors.New("please use yaml or json output format when using --config-only flag")
+	}
+	return nil
+}
+
+// initialises use colors options
+func (s *ScanOptions) initColor() {
+	switch strings.ToLower(s.useColors) {
+	case "auto":
+		if isatty.IsTerminal(os.Stdout.Fd()) || isatty.IsCygwinTerminal(os.Stdout.Fd()) {
+			s.UseColors = true
+		} else {
+			s.UseColors = false
+		}
+
+	case "true":
+		fallthrough
+	case "t":
+		fallthrough
+	case "y":
+		fallthrough
+	case "1":
+		fallthrough
+	case "force":
+		s.UseColors = true
+
+	default:
+		s.UseColors = false
+	}
+}
+
 // Run executes terrascan in CLI mode
-func Run(iacType, iacVersion string, cloudType []string,
-	iacFilePath, iacDirPath, configFile string, policyPath []string,
-	format, remoteType, remoteURL string, configOnly, useColors, verbose bool) {
+func (s *ScanOptions) Run() error {
 
 	// temp dir to download the remote repo
 	tempDir := filepath.Join(os.TempDir(), utils.GenRandomString(6))
 	defer os.RemoveAll(tempDir)
 
 	// download remote repository
-	path, err := downloadRemoteRepository(remoteType, remoteURL, tempDir)
```

Comment:
Doesn't seem like downloadRemoteRepository belongs in the scan options class... I think it could be better off separate.

Reply:
n/a